pkg/registry/service/rest.go

```diff
@@ -277,8 +277,8 @@ var _ = rest.Redirector(&REST{})
 
 // ResourceLocation returns a URL to which one can send traffic for the specified service.
 func (rs *REST) ResourceLocation(ctx api.Context, id string) (*url.URL, http.RoundTripper, error) {
-	// Allow ID as "svcname" or "svcname:port".
-	svcName, portStr, valid := util.SplitPort(id)
+	// Allow ID as "svcname", "svcname:port", or "svcname:port:scheme".
```

Comment:
the url spec allows %2f, but the golang http handler stack doesn't handle that well... it decodes it early in the stack so the escaped version is lost. Coming back to this, I prefer the scheme at the beginning as well... will update